ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -269,6 +280,37 @@ func (r *RayClusterReconciler) rayClusterReconcile(ctx context.Context, request
 	return ctrl.Result{RequeueAfter: time.Duration(requeueAfterSeconds) * time.Second}, nil
 }
 
+// Checks whether the old and new RayClusterStatus are inconsistent by comparing different fields. If the only
+// differences between the old and new status are the `LastUpdateTime` and `ObservedGeneration` fields, the
+// status update will not be triggered.
+//
+// TODO (kevin85421): The field `ObservedGeneration` is not being well-maintained at the moment. In the future,
+// this field should be used to determine whether to update this CR or not.
```

Comment:
do we have GH issue for this `ObservedGeneration` fix?

Reply:
Good question. I will open an issue for it.

Reply:
Open an issue https://github.com/ray-project/kuberay/issues/1217.